editoast/src/views/paced_train.rs

```diff
@@ -262,23 +314,53 @@ pub struct ElectricalProfileSetIdQueryParam {
 )]
 async fn simulation(
     State(AppState {
-        valkey: _valkey_client,
-        core_client: _core_client,
-        db_pool: _db_pool,
+        valkey: valkey_client,
+        core_client,
+        db_pool,
         ..
     }): State<AppState>,
-    Extension(_auth): AuthenticationExt,
-    Path(PacedTrainIdParam {
-        id: _paced_train_id,
-    }): Path<PacedTrainIdParam>,
-    Query(InfraIdQueryParam {
-        infra_id: _infra_id,
-    }): Query<InfraIdQueryParam>,
+    Extension(auth): AuthenticationExt,
+    Path(PacedTrainIdParam { id: paced_train_id }): Path<PacedTrainIdParam>,
+    Query(InfraIdQueryParam { infra_id }): Query<InfraIdQueryParam>,
     Query(ElectricalProfileSetIdQueryParam {
-        electrical_profile_set_id: _electrical_profile_set_id,
+        electrical_profile_set_id,
     }): Query<ElectricalProfileSetIdQueryParam>,
 ) -> Result<Json<SimulationResponse>> {
-    todo!();
+    let authorized = auth
+        .check_roles([BuiltinRole::OperationalStudies].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Forbidden.into());
+    }
+
+    // Retrieve infra or fail
+    let infra = Infra::retrieve_or_fail(&mut db_pool.get().await?, infra_id, || {
+        PacedTrainError::InfraNotFound { infra_id }
+    })
+    .await?;
+
+    // Retrieve paced_train or fail
+    let paced_train =
+        PacedTrain::retrieve_or_fail(&mut db_pool.get().await?, paced_train_id, || {
+            PacedTrainError::NotFound { paced_train_id }
+        })
+        .await?;
+
+    // Compute simulation of a paced_train
+    let (simulation, _) = train_simulation_batch(
+        &mut db_pool.get().await?,
+        valkey_client,
+        core_client,
+        &[paced_train.into()],
```

Comment:
nitpick: I often don't like too much `.into()` since it hides the "into what" (you'll have to look somewhere else to figure out the types). You can use the `From` instead, which makes things more obvious. What do you think?

```suggestion
        &[TrainSchedule::from(paced_train)],
```